config-model/src/main/java/com/yahoo/searchdefinition/ImmutableSearch.java

```diff
@@ -15,11 +19,22 @@
  */
 public interface ImmutableSearch {
 
+    String getName();
+    Index getIndex(String name);
+    SDField getConcreteField(String name);
```

Comment:
Consider changing the return type to `ImmutableSDField`

Reply:
Will do in a follow up PR